Cogs/Helpers/websiteHelpers.py

```diff
@@ -0,0 +1,312 @@
+            except ValueError:
+                continue
+            return True
+
+        return False
+
+    async def get_all_teams(self, game:str) -> list[Team]:
+        await self.update_cache()
+
+        return self.teams[game]
+
+    """
+    If game is specified, returns a team with that id in that game
+    Otherwise searches through all games
+    returns None if no game matches the id
+    """
+    async def get_by_id(self, id:int, game:str = None) -> Union[Team, None]:
+        await self.update_cache()
+        if(game != None):
+            return utils.get(self.teams[game], id=id)
+
+        for gameName in GameNames:
+            search = utils.get(self.teams[gameName], id=id)
+            if search != None:
+                return search
+
+        return None
+
+    """
+    If game is specified, returns a team by that name in that game
+    Otherwise searches through all games
+    returns None if no game matches the name
+    """
+    async def get_by_name(self, name:str, game:str = None) -> Union[Team, None]:
+        await self.update_cache()
+        if(game != None):
+            return utils.get(self.teams[game], name=name)
+
+        for gameName in GameNames:
+            search = utils.get(self.teams[gameName], name=name)
+            if search != None:
+                return search
+
+        return None
+
+    """
+    Attempts to create a Team.
+
+    If the server accepts the creation, the team is added to the cache and None is returned
+    Otherwise the failure message is returned.
+    """
+    async def create_team(self, name: str, game: str) -> Union[str, None]:
+        # Cache duplicate checking
+        if (await self.get_by_name(name, game)) != None:
+            return "A team with that name already exists!"
+
+        data = {
+            "Name": name,
+            "Game": game
+        }
+        status, response = await sendPost("NewTeam", data)
+
+        if(status == 200):
+            print(f"{game} team created: {name}")
+            team = Team({"id":response["id"], "name": name})
+            self.teams[game].append(team)
+            return None
+        else:
+            if(response['message'] == 'Duplicate Team Name'):
+                return "A team with that name already exists!" #TODO DUPLICATE MESSAGE
+            else:
+                return "Failed to create team"
+
+    """
+    Attempts to delete a Team.
+
+    If the server accepts the deletion, the team is removed from the cache and None is returned
+    Otherwise the failure message is returned.
+
+    Must specify at least id or name, ideally name and game
+    """
+    async def delete_team(self, id:int = None, name: str = None, game: str = None) -> Union[str, None]:
+        # Check it exists
+        team = None
+        if id != None:
+            team = await self.get_by_id(id, game)
+            if team == None:
+                return "Cannot find a team with that id"
+        elif name != None:
```

Comment:
Same possible problem with team name here as previous comment

Reply:
Again I think this was a worst case scenario where I was considering too many possible cases. I'll go through the flow and make sure this can never happen.